### ISSUE TITLE
Modifying LDAP_ROOT value

### ISSUE THREAD

**Issue by vaikundamv** [bitnami/bitnami-docker-openldap#4]
Problem:
 Changing LDAP_ROOT env to any other value ( for example, dc=example1, dc=org )

docker-compose.yaml
---------------------
version: '2'

services:
  openldap:
    image: bitnami/openldap:latest
    ports:
      - '1389:1389'
      - '1636:1636'
    environment:
      LDAP_ADMIN_USERNAME: "admin"
      LDAP_ADMIN_PASSWORD: "admin"
      LDAP_USERS: "user01,user02,user03"
      LDAP_PASSWORDS: "password1,password2,password3"
      LDAP_USER_DC: "Groups"
      LDAP_ROOT: "dc=example,dc=org"
    volumes:
      - 'openldap_data:/bitnami/openldap'

volumes:
  openldap_data:
    driver: local

----------------------

Steps:
  I tried to setup the default bitnami image (with LDAP_ROOT as dc=example, dc=org)
  Container was up with the first attempt ( docker-compose up -d ) 

 I can perform search as below where I can see all the users

ldapsearch  -h localhost -p 1389 -D cn="admin,dc=example,dc=org" -w admin  -b "dc=example,dc=org" objectclass=*

Problem that I noticed: 
 When I switched the LDAP_ROOT to something like "dc=example1, dc=com" in the docker-compose.yaml
 I noticed couple of things.

1)docker-compose up -d doesn't bring up any container at the first attempt
 I have to run it again to bring up the container

2) when I tried the search as below
ldapsearch  -h localhost -p 1389 -D cn="admin,dc=example1,dc=org" -w admin  -b "dc=example1,dc=org" objectclass=*

it returns 
search: 2
result: 32 No such object
-------------------

I am not sure if I am missing something. Please let me know if I am missing anything.
I am a beginner and kindly excuse if I missed something trivial. Thanks for your help in advance

To summarize, I was trying the following
Change to a different LDAP_ROOT (dc=example1, dc=org) and perfrom ldapsearch for the users using the below command by exec into the container

ldapsearch  -h localhost -p 1389 -D cn="admin,dc=example1,dc=org" -w admin  -b "dc=example1,dc=org" objectclass=*

Thanks

**Comment by vaikundamv** [comment]
Found this while trouble shooting

ldap_create_tree under libopenldap.sh expects dc as example as shown below.
Is this intended? 

# Root creation
dn: $LDAP_ROOT
objectClass: dcObject
objectClass: organization
dc: example
o: example

**Comment by joancafom** [comment]
Hi @vaikundamv !

Thanks so much for your issue, I will try to reproduce it and get back to you on my findings.

**Comment by vaikundamv** [comment]
@joancafom 

Thanks for looking into this .

**Comment by vaikundamv** [comment]
What I mentioned above seem to be the issue.
I was able to work around with a custom Dockerfile with the below updates to switch properly to "dc=example1,dc=org".
It would be nice if "dc" and "o" attributes are picked up from LDAP_ROOT or an env variable rather than pointing to "example"

RUN sed -i 's/dc: example/dc: example1/g' /opt/bitnami/scripts/libopenldap.sh
RUN sed -i 's/o: example/o: example1/g' /opt/bitnami/scripts/libopenldap.sh

**Comment by joancafom** [comment]
Hi @vaikundamv 

Thanks for your comment! I was successfully able to reproduce the issue. I will work on fixing that and provide with a PR soon.

Thank you!

**Comment by joancafom** [comment]
Hey @vaikundamv,

Just to let you know that I have implemented some changes and they will be released soon. I will keep this thread updated to let you know when they are out!

**Comment by joancafom** [comment]
[Changes](https://github.com/bitnami/bitnami-docker-openldap/commit/8441cf4e8884a121540d7aa4f4d860b4a5b7612b) were implemented in the latest release!

Thanks for your issue!!

**Comment by vaikundamv** [comment]
Thank you for fixing this issue
